kgforge/core/forge.py

```diff
@@ -283,7 +283,8 @@ def template(
     def validate(
         self,
         data: Union[Resource, List[Resource]],
-        execute_actions_before: bool = False,
+        execute_actions_before: bool=False,
+        type_=None
```

Comment:
I recommend to have 'None' as a default value only here at forge level. The downstream methods can just have 'type_' without default value.

Reply:
@MFSY By downstream methods you mean the methods that are not in `forge.py`?

Reply:
Yes.  this method for example call a private method from _model.

Reply:
I changed the defaults to be None only for `.validate()` in `forge.py`. All other methods get the value downstream from `validate`